jmxtrans-output/jmxtrans-output-core/src/test/java/com/googlecode/jmxtrans/model/output/support/GraphiteWriterFactoryTest.java

```diff
@@ -178,6 +180,83 @@ public void writer_using_udp_protocol() throws LifecycleException, URISyntaxExce
 		}
 	}
 
+	@Test
+	public void socketExpirationIsUsedByDefault() throws LifecycleException, URISyntaxException {
+		ImmutableList<Server> servers = configurationParser.parseServers(ImmutableList.of(file("/graphite-writer-factory-example2.json")), false);
+
+		assertThat(servers).hasSize(1);
+		Server server = servers.get(0);
+
+		assertThat(server.getQueries()).hasSize(1);
+		Query query = server.getQueries().iterator().next();
+
+		assertThat(query.getOutputWriterInstances()).hasSize(1);
+		OutputWriter outputWriter = query.getOutputWriterInstances().iterator().next();
+
+		assertThat(outputWriter).isInstanceOf(ResultTransformerOutputWriter.class);
+
+		ResultTransformerOutputWriter resultTransformerOutputWriter = (ResultTransformerOutputWriter) outputWriter;
+		OutputWriter target = resultTransformerOutputWriter.getTarget();
+
+		assertThat(target).isInstanceOf(WriterPoolOutputWriter.class);
+
+		LifecycledPool writerPool = ((WriterPoolOutputWriter) target).getWriterPool();
+
+		assertThat(writerPool).isInstanceOf(BlazePool.class);
+
+		BlazePool blazePool = (BlazePool) writerPool;
+
+		try {
+			Field expirationField = blazePool.getClass().getDeclaredField("deallocRule");
```

Comment:
This assertion is really just a rewrite of the actual code. An integration test which make sure that the connections are dropped after expiration would make more sense.

Reply:
Added integration test for actual behaviour. Here we're testing the factory, so I guess it's ok to check if everything is assembled correctly.